src/opm/output/eclipse/AggregateWellData.cpp

```diff
@@ -0,0 +1,773 @@
+                    .depthTopSegment();
+            }
+
+            // Not a multi-segment well--i.e., this is a regular
+            // well.  Use well's reference depth.
+            return well.getRefDepth(sim_step);
+        }
+
+        Opm::RestartIO::Helpers::WindowedArray<float>
+        allocate(const std::vector<int>& inteHead)
+        {
+            using WV = Opm::RestartIO::Helpers::WindowedArray<float>;
+
+            return WV {
+                WV::NumWindows{ numWells(inteHead) },
+                WV::WindowSize{ entriesPerWell(inteHead) }
+            };
+        }
+
+        std::vector<float> defaultSWell()
+        {
+            const auto dflt  = -1.0e+20f;
+            const auto infty =  1.0e+20f;
+            const auto zero  =  0.0f;
+            const auto one   =  1.0f;
+            const auto half  =  0.5f;
+
+            // Initial data by Statoil ASA.
+            return { // 122 Items (0..121)
+                // 0     1      2      3      4      5
+                infty, infty, infty, infty, infty, infty,    //   0..  5  ( 0)
+                one  , zero , zero , zero , zero , 1.0e-05f, //   6.. 11  ( 1)
+                zero , zero , infty, infty, zero , dflt ,    //  12.. 17  ( 2)
+                infty, infty, infty, infty, infty, zero ,    //  18.. 23  ( 3)
+                one  , zero , zero , zero , zero , zero ,    //  24.. 29  ( 4)
+                zero , one  , zero , infty, zero , zero ,    //  30.. 35  ( 5)
+                zero , zero , zero , zero , zero , zero ,    //  36.. 41  ( 6)
+                zero , zero , zero , zero , zero , zero ,    //  42.. 47  ( 7)
+                zero , zero , zero , zero , zero , zero ,    //  48.. 53  ( 8)
+                infty, zero , zero , zero , zero , zero ,    //  54.. 59  ( 9)
+                zero , zero , zero , zero , zero , zero ,    //  60.. 65  (10)
+                zero , zero , zero , zero , zero , zero ,    //  66.. 71  (11)
+                zero , zero , zero , zero , zero , zero ,    //  72.. 77  (12)
+                zero , infty, infty, zero , zero , one  ,    //  78.. 83  (13)
+                one  , one  , zero , infty, zero , infty,    //  84.. 89  (14)
+                one  , dflt , one  , zero , zero , zero ,    //  90.. 95  (15)
+                zero , zero , zero , zero , zero , zero ,    //  96..101  (16)
+                zero , zero , zero , zero , zero , zero ,    // 102..107  (17)
+                zero , zero , half , one  , zero , zero ,    // 108..113  (18)
+                zero , zero , zero , zero , zero , infty,    // 114..119  (19)
+                zero , one  ,                                // 120..121  (20)
+            };
+        }
+
+        template <class SWellArray>
+        void assignDefaultSWell(SWellArray& sWell)
+        {
+            const auto& init = defaultSWell();
+
+            const auto sz = static_cast<
+                decltype(init.size())>(sWell.size());
```

Comment:
This reduces readability, and likely does not give any protection from errors. I struggle to see what could go wrong with `std::min(init.size(), sWell.size())`, or using plain `int` here.